datahub-frontend/app/controllers/Application.java

```diff
@@ -73,13 +99,35 @@ public Result index(@Nullable String path) {
   }
 
   /**
-   * Generic not found response
-   * @param path
-   * @return
+   * Proxies requests to the Metadata Service
+   *
+   * TODO: Investigate using mutual SSL authentication to call Metadata Service.
    */
-  @Nonnull
-  public Result apiNotFound(@Nullable String path) {
-    return badRequest("{\"error\": \"API endpoint does not exist\"}");
+  @Security.Authenticated(Authenticator.class)
+  public CompletableFuture<Result> proxy(String path) throws ExecutionException, InterruptedException {
+    final String resolvedUri = PATH_REMAP.getOrDefault(request().uri(), request().uri());
+    return _ws.url(String.format("http://%s:%s%s", GMS_HOST, GMS_PORT, resolvedUri))
```

Comment:
will this always be http?

Reply:
See comment above^ in order to serve mutual auth, we'll have to do some additional client side configuration work. I don't want to include that here. Think I should create a ticket?